docs/user-guide/pod-preset/index.md

```diff
@@ -0,0 +1,543 @@
+---
+assignees:
+- jessfraz
+title: Pod Preset
+---
+
+You can use a `podpreset` object to inject certain information into pods at creation
+time. This information can include secrets, volumes, volume mounts, and environment
+variables.
+
+See [PodPreset proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-preset.md) for more information.
+
+* TOC
+{:toc}
+
+## What is a Pod Preset?
+
+A _Pod Preset_ is an API resource that you can use to inject additional runtime
+requirements into a Pod at creation time. You use label selectors to specify
+the Pods to which a given Pod Preset applies.
+
+Using a Pod Preset allows pod template authors to not have to explicitly set
+information for every pod. This way, authors of pod templates consuming a
+specific service do not need to know all the details about that service.
+
+## Admission Control
+
+_Admission control_ is how Kubernetes applies Pod Presets to incoming pod
+creation requests. When a pod creation request occurs, the system does the
+following:
+
+1. Retrieve all `PodPresets` available for use.
+1. Match the label selector of the `PodPreset` to the pod being created.
+1. Attempt to merge the various defined resources for the `PodPreset` into the
+   Pod being created.
+1. On error, throw an event documenting the merge error on the pod, and create
+   the pod _without_ any injected resources from the `PodPreset`.
+
+### Behavior
+
+When a `PodPreset` is applied to one or more Pods, Kubernetes modifies the pod
+spec. For changes to `Env`, `EnvFrom`, and `VolumeMounts`, Kubernetes modifies
+the container spec for all containers in the Pod; for changes to Volume,
+Kubernetes modifies the Pod Spec.
+
+Kubernetes annotates the resulting modified pod spec to show that it was
+modified by a `PodPreset`. The annotation is of the form
+`podpreset.admission.kubernetes.io/<pip name>": "<resource version>"`.
+
+
+## Enabling Pod Preset
+
+In order to use Pod Presets in your cluster you must ensure the
+following
+
+1.  You have enabled the api type `settings.k8s.io/v1alpha1/podpreset`
+1.  You have enabled the admission controller `PodPreset`
+1.  You have defined your pod presets
+
+## Creating a Pod Preset
+
+### Simple Pod Spec Example
+
+This is a simple example to show how a Pod spec is modified by the Pod
+Injection Policy.
+
+**User submitted pod spec:**
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: website
+  labels:
+    app: website
+    role: frontend
+spec:
+  containers:
+    - name: website
+      image: ecorp/website
+      ports:
+        - containerPort: 80
+```
+
+**Example Pod Preset:**
+
+```yaml
+kind: PodPreset
+apiVersion: settings.k8s.io/v1alpha1
+metadata:
+  name: allow-database
+  namespace: myns
+spec:
+  selector:
```

Comment:
Added a link to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors